test/e2e/data/e2e_conf.yaml

```diff
@@ -128,9 +128,9 @@ providers:
 variables:
   # used to ensure we deploy to the correct management cluster
   KUBE_CONTEXT: "kind-capo-e2e"
-  KUBERNETES_VERSION: "v1.25.0"
-  KUBERNETES_VERSION_UPGRADE_FROM: "v1.24.9"
-  KUBERNETES_VERSION_UPGRADE_TO: "v1.25.0"
+  KUBERNETES_VERSION: "v1.26.5"
+  KUBERNETES_VERSION_UPGRADE_FROM: "v1.25.10"
+  KUBERNETES_VERSION_UPGRADE_TO: "v1.26.5"
   ETCD_VERSION_UPGRADE_TO: "3.5.4-0"
```

Comment:
Kubernetes v1.26.5 uses etcd v3.5.5, at least when checking the `go.mod`. Not sure if that is a great source of truth :)

https://github.com/kubernetes/kubernetes/blob/890a139214b4de1f01543d15003b5bda71aae9c7/go.mod#L69

Reply:
Cluster API tests upgrading to v1.26 with etcd v3.5.6
https://github.com/kubernetes/test-infra/blob/8d3c72bc9953d935ce31f109e004e25ec03c4f47/config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-4-upgrades.yaml#L239-L240

Reply:
By the way, could we use  `stable-v1.25` and  `stable-v1.26` here as CAPI does it?
https://github.com/kubernetes/test-infra/blob/8d3c72bc9953d935ce31f109e004e25ec03c4f47/config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-4-upgrades.yaml#L235-L238

Reply:
Cool! I was not aware that this was possible. For now we actually have the upgrade test disabled because we use a plain cloud image. Since we use `preKubeadmCommands` to install kubelet, kubeadm etc. we cannot "switch" version in the middle of the test. :disappointed: 

I have it on my TODO to build images that we could use for these tests so we can enable them again.
I'm curious how CAPI makes sure the necessary images are always available? :thinking:

Reply:
Ok now I see what they do. They have additional logic in their scripts that resolves these versions and builds the relevant images before starting the test.
https://github.com/kubernetes-sigs/cluster-api/blob/75c29acbb784ce26daf05d670d949eb44f76b439/scripts/ci-e2e-lib.sh#L81

Reply:
ETCD version bumped, but I left the other versions as they were since we don't have the fancy script that CAPI use to translate `stable-v1.25` and the like. :slightly_smiling_face: